### PR TITLE
feat: group chat empty view created by other user 

### LINKED ISSUE #272: [M1] `GroupChatEmptyView` is wrong for member of group
Image

### PR DESCRIPTION
**Pull request by imGok**
emptyview

This Closes #272

### CODE REVIEW

**Review by hoangdat — APPROVED** [review]

File: lib/pages/chat/group_chat_empty_view.dart [hunk 1]
```diff
@@ -1,67 +1,131 @@
...
 
   @override
   Widget build(BuildContext context) {
+    final hasCreatedRoom = firstEvent.type == EventTypes.RoomCreate &&
+      firstEvent.senderId == Matrix.of(context).client.userID;
+
+    return hasCreatedRoom ? _buildOwnGroupChatEmptyView(context) : _buildOthersGroupChatEmptyView(context, firstEvent);
+  }
+
+  Container _buildOthersGroupChatEmptyView(BuildContext context, Event firstEvent) {
```
Comment (hoangdat) [inline lib/pages/chat/group_chat_empty_view.dart:24]: should be a widget

File: lib/pages/chat/group_chat_empty_view.dart [hunk 2]
```diff
@@ -1,67 +1,131 @@
...
-          const SizedBox(height: 26.0,),
-          Text(
-            L10n.of(context)!.youCreatedGroupChat,
-            style: TextStyle(
-              fontWeight: FontWeight.w700,
-              fontSize: 17,
-              color: Theme.of(context).colorScheme.onSurface,
-            ),
-          ),
+          _getDateWidget(context, firstEvent),
```
Comment (hoangdat) [inline lib/pages/chat/group_chat_empty_view.dart:32]: Why not keep the logic to show the date out of the `empty view`?

**Review by hoangdat — APPROVED** [review]

**Review by hoangdat — CHANGES_REQUESTED** [review]
srr, but fix the above comments. Thanks

File: lib/pages/chat/group_chat_empty_view.dart (hunk 2 above)
Comment (imGok) [reply lib/pages/chat/group_chat_empty_view.dart:32]: On the mockup we don't show date when it's our own group 
image

Comment (imGok) [reply lib/pages/chat/group_chat_empty_view.dart:32]: these 2 empty groups views wont stay if there is a message or an event, it's a placeholder

**Review by drminh2807 — APPROVED** [review]

**Review by sherlockvn — APPROVED** [review]

**Comment by imGok** [comment]
updated demo
emptygrp

**Review by hoangdat — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**